tools/travis/stage.sh

```diff
@@ -126,6 +127,10 @@ flink-connectors/flink-sql-connector-kafka,"
 MODULES_TESTS="\
 flink-tests"
 
+MODULES_SCHEDULR_NG="\
+$MODULES_CORE,\
+$MODULES_TESTS"
```

Comment:
If the tests are fast, I would find it better to run the tests in core modules as part of the pre-commit tests.

Reply:
Will take a try in a following commit to see if the time cost is accepted.

Reply:
Just realized that to have an accurate measurement for stage time spent, we need #9901 (annotate core modules' tests) and #9900 (it annotates AbstractTestBase which also enables tests in flink-streaming-scala which is one of the core modules) merged first.

Reply:
I'd recommend to fully remove the annotation that you're adding and simply setting the scheduler-ng property in the scala 2.12 profile. The scheduler is scala independent anyway so that shouldn't be a problem.

That way we neither have to deal with annotations, nor the resulting additional stages and maven invocations.

Reply:
Hi @zentol ,
For the cron job, setting scheduler-ng along with scala-2.12 is an option to avoid adding new stages. 
But this may not work for jdk11 which should be verified for both legacy scheduler and ng scheduler. 

For pre-commit test, if to support fast core modules' ng scheduler testing and keep legacy scheduler testing at the same time, we may still need a new stage. Yet before adding it, we need first to check whether the time cost of this new stage is acceptable.

The annotation is used to suppress failing tests on ng scheduler(by not annotating it). Some tests cannot pass at the moment(those related issues in FLINK-14366). It also helps to avoid run non-MiniCluster tests multiple times if the new ng_scheduler/ng_scheduler_core stage is added.

Reply:
yeah that was a dumb suggestion by me at this stage; but we can revisit it once all tests are passing.

Do we have an estimate for how many tests _in total_ we need for verifying the functionality of the scheduler (including any tests that currently fail)? What's the long-term plan here, enable tests one by one until everything passes, then discard the old scheduler? Is the old one kept as a backup that users can revert to?

Reply:
The tests should be all MiniCluster/IT tests in core modules(as defined in the stage.sh) and flink-tests module.
I didn't count it because some tests are enabled through annotating its test base.
Currently annotating on core modules' tests is merged. The annotating for flink-tests is pending for review at #9900. With #9900 merged the annotation work can be done except for 4 failing tests linked in FLINK-14366.

We plan to keep the legacy scheduler as a backup in Flink 1.10 and will remove it in Flink 1.11.
When all tests pass (and also with some more stability testing to gain confidence), we can make the ng scheduler to be default. The annotation will be renamed to `AlsoRunWithLegayScheduler` and help to guarantee the legacy scheduler does not break by any further changes in Flink 1.10.

Reply:
so basically, at some point we may have a large amount of tests running against both schedulers. 

In that case we should use separate stages right off the bat. I'm fine with running both of the new stages per commit; we currently have the capacity for it.

Reply:
@zentol As you also suggested we remove the tests for JDK11. So maybe we just add the scheduler_ng stage for per commit tests. And no need to add new cron jobs? (scheduler_ng_core stage is also not needed then.)

Reply:
Here's a reference for the time cost https://travis-ci.com/zhuzhurk/flink/builds/132499016.
I annotated all passing tests and enabled full scheduler_ng and core scheduler_ng job for PRs.
![image](https://user-images.githubusercontent.com/5869249/67086897-64eb1700-f1d4-11e9-8489-039ade6b79a2.png)

Note that it's on my own travis which is much slower than the flink-ci travis. 
The `tests` job time cost can be a standard for comparing. It is usually ~16min in flink-ci travis and ~42min in my travis.

Reply:
I would add 2 stages, one `scheduler_ng_core` (MODULES_CORE) and `scheduler_ng_tests` (MODULES_TESTS). Realistically we will hit a timeout trying to run both core and tests in one stage.

Reply:
I see the concerns. 
Updated the PR to add two stages `scheduler_ng_core` and `scheduler_ng_tests` in per commit tests.